spec/unit/resource/chocolatey_installer_spec.rb

```diff
@@ -0,0 +1,109 @@
+      <chocolatey xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+        <config>
+          <add key="containsLegacyPackageInstalls" value="true" description="Install has packages installed prior to 0.9.9 series." />
+        </config>
+        <sources>
+          <source id="chocolatey" value="https://chocolatey.org/api/v2/" disabled="false" bypassProxy="false" selfService="false" adminOnly="false" priority="0" />
+        </sources>
+        <features>
+          <feature name="checksumFiles" enabled="true" setExplicitly="false" description="Checksum files when pulled in from internet (based on package)." />
+        </features>
+        <apiKeys />
+      </chocolatey>
+    CONFIG
+  end
+
+  # we save off the ENV and set ALLUSERSPROFILE so these specs will work on *nix and non-C drive Windows installs
+  before(:each) do
+    @original_env = ENV.to_hash
+    ENV["ALLUSERSPROFILE"] = "C:\\ProgramData"
+  end
+
+  after(:each) do
+    ENV.clear
+    ENV.update(@original_env)
+  end
+
+  describe "basic chocolatey settings" do
+    it "has a resource name of :chocolatey_installer" do
+      expect(resource.resource_name).to eql(:chocolatey_installer)
+    end
+
+    it "sets the default action as :install" do
+      expect(resource.action).to eql([:install])
+    end
+
+    it "supports :install and :uninstall actions" do
+      expect { resource.action :install }.not_to raise_error
+      expect { resource.action :uninstall }.not_to raise_error
+    end
+  end
+
+  describe "installing chocolatey" do
+    it "can install Chocolatey with parameters" do
+      expect { resource.action :install }.not_to raise_error
+    end
+
+    it "returns false if a chocolatey install cannot be found" do
+      allow(::File).to receive(:exist?).with('C:\ProgramData\chocolatey\bin\choco.exe').and_return(false)
+      expect(resource.fetch_choco_installer).to eql(false)
+    end
+  end
+
+  describe "chocolatey is idempotent because it" do
+    it "does not install choco again if it is already installed" do
+      install_choco
+      chocolatey_installer "install" do
+        action :install
+      end.should_not_be_updated
+    end
+  end
+
+  describe "uinstalling chocolatey" do
+    it "doesn't error out uninstalling chocolatey if chocolatey is not installed" do
+      allow(::File).to receive(:exist?).with('C:\ProgramData\chocolatey\bin\choco.exe').and_return(false)
+      expect { resource.action :uninstall }.not_to raise_error
+    end
+  end
+
+  def install_choco
+    include ChefPowerShell::ChefPowerShellModule::PowerShellExec
+    powershell_code = <<-CODE
+      Set-ExecutionPolicy Bypass -Scope Process -Force;
+      [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072;
+      iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+    CODE
+    powershell_exec(powershell_code)
+  end
```

Comment:
This feels tricky to me. Maybe this should be like a kitchen test or something?

Reply:
simplified